src/nn_manager.cpp

```diff
@@ -1 +1,260 @@
-#include "stdafx.h"
+#include "nn_manager.h"
+
+nn_manager::nn_manager(int time_interval,
+    int inputs_count_per_parameter) : time_interval(time_interval),
+    p_plant(new plant(1, 4, 0.5)),
+    p_pid(new PID()),
+    nn2_emulator(new mlp(20, 10, 1, 120)),
+    use_learning(true),
+    nn1_tuner(new mlp(20, 10, 3, 120))
+{
+    //Создание необходимых массивов и выделение памяти [1]. Запись
+    //данных инициализации [2]. Запись указателей для выборки обучения [3].
+    p_plant_data = new std::vector<float>();                        //1
+    PID_data = new std::vector<float>();
+
+    nn2_out = new std::vector<float>;
+    nn2_in = new std::vector<float*>;
+
+    p_plant_data->reserve(time_interval);
+    PID_data->reserve(time_interval);
+    nn2_out->reserve(time_interval);
+
+    int samples_count = time_interval - inputs_count_per_parameter + 1;
+    nn2_in->reserve(2 * inputs_count_per_parameter * samples_count);
+
+    for (int i = 0; i < time_interval; i++)                       //2
+    {
+        p_plant_data->push_back(0);
+        PID_data->push_back(0);
+        nn2_out->push_back(0);
+    }
```

Comment:
Чтобы не нарушать целостность с текущим стилем, фигурные скобки размещаем на одном уровне с кодом:
```suggestion
    for (int i = 0; i < time_interval; i++)                       //2
        {
        p_plant_data->push_back(0);
        PID_data->push_back(0);
        nn2_out->push_back(0);
        }
```